table/tables/partition_test.go

```diff
@@ -447,6 +447,25 @@ func (ts *testSuite) TestCreatePartitionTableNotSupport(c *C) {
 	c.Assert(ddl.ErrPartitionFunctionIsNotAllowed.Equal(err), IsTrue)
 }
 
+// issue 24880
+func (ts *testSuite) TestRangePartitionUnderNoUnsignedSub(c *C) {
+	tk := testkit.NewTestKitWithInit(c, ts.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists tu;")
+	tk.MustExec("SET @@sql_mode='NO_UNSIGNED_SUBTRACTION';")
+	_, err := tk.Exec(`CREATE TABLE tu (c1 BIGINT UNSIGNED) PARTITION BY RANGE(c1 - 10) (
+						PARTITION p0 VALUES LESS THAN (-5),
+						PARTITION p1 VALUES LESS THAN (0),
+						PARTITION p2 VALUES LESS THAN (5),
+						PARTITION p3 VALUES LESS THAN (10),
+						PARTITION p4 VALUES LESS THAN (MAXVALUE)
+						);`)
+	c.Assert(err, IsNil)
+	// currently not support insert records whose partition value is negative
```

Comment:
I don't know why it is not supported, can you please explain a little while?

Reply:
Currently the partition expression used to locate partitions uses a mock context instead of the current context, and thus it cannot do unsigned subtraction producing negative results under NO_UNSIGNED_SUBTRACTION SQLMode. See [source code](https://github.com/pingcap/tidb/blob/master/table/tables/partition.go#L116).